lib/streamlit/color_util.py

```diff
@@ -0,0 +1,245 @@
+
+# components go from 0 to 255, except alpha goes from 0.0 to 1.0
+# CSS uses these.
+MixedRGBAColorTuple: TypeAlias = Tuple[int, int, int, float]
+
+ColorTuple: TypeAlias = Union[
+    FloatRGBColorTuple,
+    FloatRGBAColorTuple,
+    IntRGBColorTuple,
+    IntRGBAColorTuple,
+    MixedRGBAColorTuple,
+]
+
+IntColorTuple = Union[IntRGBColorTuple, IntRGBAColorTuple]
+CSSColorStr = Union[IntRGBAColorTuple, MixedRGBAColorTuple]
+
+ColorStr: TypeAlias = str
+
+Color: TypeAlias = Union[ColorTuple, ColorStr]
+MaybeColor: TypeAlias = Union[str, Collection[Any]]
+
+
+def to_int_color_tuple(color: MaybeColor) -> IntColorTuple:
+    """Convert input into color tuple of type (int, int, int, int)."""
+    color_tuple = _to_color_tuple(
+        color,
+        rgb_formatter=_int_formatter,
+        alpha_formatter=_int_formatter,
+    )
+    return cast(IntColorTuple, color_tuple)
+
+
+def to_css_color(color: MaybeColor) -> Color:
+    """Convert input into a CSS-compatible color that Vega can use.
+
+    Inputs must be a hex string, rgb()/rgba() string, or a color tuple. Inputs may not be a CSS
+    color name, other CSS color function (like "hsl(...)"), etc.
+
+    See tests for more info.
+    """
+    if is_css_color_like(color):
+        return cast(Color, color)
+
+    if is_color_tuple_like(color):
+        ctuple = cast(ColorTuple, color)
+        ctuple = _normalize_tuple(ctuple, _int_formatter, _float_formatter)
+        if len(ctuple) == 3:
+            return f"rgb({ctuple[0]}, {ctuple[1]}, {ctuple[2]})"
+        elif len(ctuple) == 4:
+            c4tuple = cast(MixedRGBAColorTuple, ctuple)
+            return f"rgba({c4tuple[0]}, {c4tuple[1]}, {c4tuple[2]}, {c4tuple[3]})"
+
+    raise InvalidColorException(color)
+
+
+def is_css_color_like(color: MaybeColor) -> bool:
+    """Check whether the input looks like something to_css_color() can produce.
+
+    This is meant to be lightweight, and not a definitive answer. The definitive solution is to try
+    to convert and see if an error is thrown.
+    """
+    return is_hex_color_like(color) or _is_cssrgb_color_like(color)
+
+
+def is_hex_color_like(color: MaybeColor) -> bool:
+    """Check whether the input looks like a hex color.
+
+    NOTE: We only accept hex colors and color tuples as user input. So you should use
+    is_hex_color_like and is_color_tuple_like whenever checking your inputs.
+
+    This is meant to be lightweight, and not a definitive answer. The definitive solution is to try
+    to convert and see if an error is thrown.
+    """
+    return (
+        isinstance(color, str)
+        and color.startswith("#")
+        and color[1:].isalnum()  # Alphanumeric
+        and len(color) in {4, 5, 7, 9}
+    )
+
+
+def _is_cssrgb_color_like(color: MaybeColor) -> bool:
+    """Check whether the input looks like a CSS rgb() or rgba() color string.
+
+    NOTE: We only accept hex colors and color tuples as user input. So you should use
```

Comment:
Just double checking: does this note belong here?

Reply:
Improved this and similar comments.